app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -1315,7 +1331,9 @@ private void doSave() {
             return;
         }
 
-        if (!validBalance) {
+        CharSequence balanceFieldEditTextError = balanceField.getError();
+
+        if (balanceFieldEditTextError != null) {
             Snackbar.make(balanceField, getString(R.string.parsingBalanceFailed, balanceField.getText().toString()), Snackbar.LENGTH_LONG).show();
```

Comment:
This is not using the `balanceFieldEditTextError.toString()` like the rest.

Reply:
@TheLastProject , the string value **parsingBalanceFailed** has a placeholder for the value, so I figured you would want to use it. 

` <string name="parsingBalanceFailed"><xliff:g>%s</xliff:g> does not seem to be a valid balance.</string>`

If not, then we need to use a different string from the strings.xml.

Reply:
I don't understand your comment.

On line 418 you use `setError()` to put the full string in it already:
```
balanceField.setError(getString(R.string.parsingBalanceFailed, balanceField.getText().toString()));
```

So shouldn't you be possible to just do this?
```
Snackbar.make(balanceField, balanceFieldEditTextError.toString(), Snackbar.LENGTH_LONG).show();
```

Or am I missing something?

Reply:
Ah, now I see what you meant. Changed.